nomad/util.go

```diff
@@ -196,6 +196,34 @@ func shuffleStrings(list []string) {
 	}
 }
 
+// partitionAll splits a slice of strings into a slice of slices of strings, each with a max
+// size of `size`. All entries from the original slice are preserved. The last slice may be
+// smaller than `size`. The input slice is unmodified
+func partitionAll(size int, xs []string) [][]string {
+	out := make([][]string, 0)
+	if size < 1 {
+		return append(out, xs)
```

Comment:
nitpick - I suspect you can also do which I find a little easier to read but that's subjective.

```suggestion
		return [][]string{out}
```